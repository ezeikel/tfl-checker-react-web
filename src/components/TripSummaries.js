import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import styled from "styled-components";
import TripSummary from "./TripSummary";
import { setSelectedTrip } from "../redux/actions";

const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  margin-top: var(--spacing-huge);
  > span {
    &:first-of-type {
      font-size: 2.4rem;
      margin-bottom: 32px;
      font-weight: bold;
      color: var(--color-white);
    }
  }
  a {
    display: flex;
    width: 100%;
    align-items: center;
    color: var(--color-text);
    &:first-of-type {
      border-top-left-radius: 4px;
      border-top-right-radius: 4px;
      section {
        border-top-left-radius: 4px;
        border-top-right-radius: 4px;
      }
    }
    &:last-of-type {
      border-bottom-left-radius: 4px;
      border-bottom-right-radius: 4px;
      section {
        border-bottom-left-radius: 4px;
        border-bottom-right-radius: 4px;
      }
    }
    & + a {
      border-top: 1px solid #eeeeee;
    }
  }
`;

const StyledTripSummary = styled(TripSummary)`
  border-radius: 0;
  &:hover {
    background-color: #f2f2f2;
    cursor: pointer;
  }
`;

const TripSummaries = ({ journeys, onSetSelectedTrip }) => (
  <Wrapper>
    <span>Suggested</span>
    {journeys.map((journey, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <Link to="/trip" onClick={() => onSetSelectedTrip(journey)} key={i}>
        <StyledTripSummary journey={journey} />
      </Link>
    ))}
  </Wrapper>
);

const mapDispatchToProps = dispatch => ({
  onSetSelectedTrip: value => dispatch(setSelectedTrip(value)),
});

TripSummaries.propTypes = {
  journeys: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  onSetSelectedTrip: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(TripSummaries);
