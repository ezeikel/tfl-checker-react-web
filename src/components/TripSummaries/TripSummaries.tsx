import { Link } from "react-router-dom";
import { setSelectedTrip } from "../../redux/actions";
import { useAppDispatch } from "../../hooks";
import { Wrapper, StyledTripSummary } from "./TripSummaries.styled";

type TripSummariesProps = {
  journeys: any[]; // TODO:
};

const TripSummaries = ({ journeys }: TripSummariesProps) => {
  const dispatch = useAppDispatch();

  return (
    <Wrapper>
      <span>Suggested</span>
      {journeys.map((journey) => (
        // eslint-disable-next-line jsx-a11y/anchor-is-valid
        <Link
          to="/trip"
          onClick={() => dispatch(setSelectedTrip(journey))}
          key={journey.id}
        >
          <StyledTripSummary journey={journey} />
        </Link>
      ))}
    </Wrapper>
  );
};

export default TripSummaries;
