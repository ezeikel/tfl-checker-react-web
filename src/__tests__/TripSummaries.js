import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import TripSummaries from '../components/TripSummaries';

const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);

let mockProps;

beforeEach(() => {
  mockProps = {
    journeys: [],
    onSetSelectedTrip: jest.fn(),
  };
});

test('renders without crashing', () => {
  const store = mockStore();

  const { container } = render(
    <TripSummaries {...mockProps} store={store} /> // eslint-disable-line react/jsx-props-no-spreading
  );
  expect(container).toMatchSnapshot();
});
