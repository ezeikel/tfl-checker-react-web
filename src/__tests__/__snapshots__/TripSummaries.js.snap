// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <section
    class="sc-AxgMl hrVdCS"
  >
    <span>
      Suggested
    </span>
  </section>
</div>
`;
